frontend/src/components/adminTable/table.tsx

```diff
@@ -63,39 +143,62 @@ export default function AdminTable({ users }: { users: User[] }) {
           ))}
         </tbody>
       </table>
-      {users.length === 0 && (
+      {sortedUsers.length === 0 && (
         <h1 className={styles.emptyTable}>Could not display any users</h1>
       )}
-      {settingsModal >= 0 && (
+      {settingsUser && (
         <Modal
           setShowModal={closeModal}
-          title={`Settings for ${users[settingsModal].name}`}
+          title={`Settings for ${settingsUser.name}`}
           content={
             <div className={styles.settingsModal}>
               <div className={styles.settingsInputContainer}>
-                <div>Name:</div>
+                <div className={styles.settingsInputLabel}>Name:</div>
                 <input
                   className={styles.settingsInput}
                   type="text"
-                  value={users[settingsModal].name}
+                  value={settingsUser.name}
+                  onChange={(e) =>
+                    handleSettingsTextChange(
+                      settingsUser,
+                      "name",
+                      e.target.value,
+                    )
+                  }
                 />
-                <div>Email:</div>
+                <div className={styles.settingsInputLabel}>Email:</div>
                 <input
                   className={styles.settingsInput}
                   type="text"
-                  value={users[settingsModal].email}
+                  value={settingsUser.email}
+                  onChange={(e) =>
+                    handleSettingsTextChange(
+                      settingsUser,
+                      "email",
+                      e.target.value,
+                    )
+                  }
+                />
+                <div className={styles.settingsInputLabel}>Roles:</div>
+                <SearchDropdown
+                  options={employeeRoles}
+                  placeholder="Select Role..."
+                  subheader=""
+                  isMulti
                 />
               </div>
               <div className={styles.settingsButtonGroup}>
-                <button className={styles.settingsButton}>Update role</button>
                 <button className={styles.settingsButton}>
-                  Revoke & Regenerate credentials
+                  Revoke credentials
```

Comment:
nit: `Credentials` should maybe be capitalized here for consistency. You seem to capitalize the first letter in each word for every other button.